### ISSUE TITLE
Empty gaps with st_mosaic

### ISSUE THREAD

**Issue by elaliberte** [r-spatial/stars#274]
I'm getting empty gaps in a mosaic I'm trying to make with `st_mosaic()`.
I understand that this is due to the way missing values are handled, but despite trying different options I can't get it to work.
Below is a reproducible example with two rasters I'm trying to mosaic together.
Any help would be greatly appreciated. Thanks.
The data can be downloaded from [here](https://drive.google.com/file/d/1oNcBf6Su9DbphCqKJswz9LRsU6SzPDch/view?usp=sharing) (150 MB).

```{r}
a <- read_stars('a.tif', proxy = T)
b <- read_stars('b.tif', proxy = T)

plot(a, reset = F)
plot(b, add = T) # shows that they are contiguous, no gap

ab <- st_mosaic(a, b)

plot(ab) # empty gap because missing values in each image hide the data
```

**Comment by elaliberte** [comment]
Here's a simpler example. The second raster listed in `st_mosaic()` hides values in the first one. This is ok, but I expected the default behaviour to be that NAs would be transparent, such that if the first (bottom) layer has data for an area in which the top layer has NAs, we would see the data from the first layer in the resulting mosaic instead of the NAs of the top layer.

```{r}
# stars example for st_mosaic
x = read_stars(system.file("tif/L7_ETMs.tif", package = "stars"))
x1 = x[,100:200,100:200, 1]
x2 = x[,150:300,150:300, 1]
plot(st_mosaic(x1, x2))

# now let's add some NA in corners of each image
x1_NA <- x1
x2_NA <- x2

x1_NA[[1]][85:101,85:101,] <- NA
x2_NA[[1]][1:15,1:15,] <- NA 

plot(x1_NA)
plot(x2_NA)

plot(st_mosaic(x1_NA, x2_NA))
plot(st_mosaic(x2_NA, x1_NA))
```

**Comment by elaliberte** [comment]
I finally found a way to solve this. This works:
```{r}
plot(st_mosaic(x2_NA, x1_NA, options = c("-srcnodata", "nan")))
```
To figure this one out, I had to first save the `stars` object as GeoTiffs
```{r}
write_stars(x1_NA, 'x1_NA.tif')
write_stars(x2_NA, 'x2_NA.tif')
```
Then opened them `L7_ETMs.tif` in QGIS. There, I found that missing values were stored as `nan`.
It took me a long time to figure this one out, and I can imagine that other future users might too.
Could this be made simpler?
